pkg/validation/component.go

```diff
@@ -212,9 +230,24 @@ func needDBUserUpdate(user api.UserRepresentation) bool {
 }
 
 func (c *component) CreateCheck(ctx context.Context, userID string, check api.CheckRepresentation) error {
+	var accessToken string
+	var err error
+
+	if check.IsIdentificationSuccessful() {
```

Comment:
I think we should prevent a scenario where we have an accreditation without persisting the check. Maybe we should swap between persistency of accrediation and check ? WHat do you think about it ?